tests/utils/async.nim

```diff
@@ -0,0 +1,25 @@
+import chronos/futures, stew/results, chronos
+import ./futures
+
+proc toOk(future: Future[void]): Result[void, string] =
+  return results.ok()
+
+proc toOk[T](future: Future[T]): Result[T, string] =
+  return results.ok(future.read())
+
+proc toResult*[T](future: Future[T]): Result[T, string] =
+  if future.cancelled():
+    return results.err("Future cancelled/timed out.")
+  elif future.finished():
+    if not future.failed():
+      return future.toOk()
+    else:
+      return results.err("Future finished but failed.")
+  else:
+    return results.err("Future still not finished.")
+
+proc waitForResult*[T](
+    future: Future[T], timeout = DURATION_TIMEOUT
+): Future[Result[T, string]] {.async.} =
+  discard await future.withTimeout(timeout)
+  return future.toResult()
```

Comment:
This complexity isn't necessary. `Result` is a substitute for aa `Exception`, not a `Future`. Our focus should be to make the codebase simpler and easier to understand, not the other way around.

Reply:
`Result` is not a substitute for `Exception`, but a more general any failed computation.
In this case, this simplifies and makes more robust a lot of checks in tests. When you run `withTimeout` you are effectively collapsing the "uncertainty of the future", that is, making it either have a value, or an error. 
Casting that "collapsed future" into a result allows for a more visual and easier handling of that situation.